libs/common/utilities/src/lib/util/utils.ts

```diff
@@ -146,3 +146,14 @@ export const returnNotEmptyString = (s: string) => (s && s !== '' ? s : undefine
 export const returnNumber = (n: number) => (!isNaN(n) ? n : undefined);
 export const ifDefinedString = (s: string) => ifDefined(returnNotEmptyString(s));
 export const ifDefinedNumber = (n: number) => ifDefined(returnNumber(n));
+
+export const findDeepestElementThroughShadowRoot = (rootElement: Element | ShadowRoot | null, selector: string): Element | null => {
+    if (rootElement) {
+        const deepestElement = Array.from(rootElement.querySelectorAll('*'))
+            .filter((el) => !!el.shadowRoot)
+            .map((el) => findDeepestElementThroughShadowRoot(el.shadowRoot, selector))
+            .find((el) => !!el);
+        return deepestElement ? deepestElement : rootElement.querySelector(selector);
```

Comment:
eventueel hier:
`deepestElement || rootElement.querySelector(selector)`
maar dat is een beetje dichterlijke vrijheid

Reply:
Nee klopt, ik vind uw voorstel ook properder. Had gewoon niet meer refactored na het 'spelen' :)